src/relax/op/tensor/manipulate.cc

```diff
@@ -1357,7 +1370,8 @@ TVM_REGISTER_OP("relax.cumsum")
     .set_attrs_type<CumsumAttrs>()
     .set_num_inputs(1)
     .add_argument("data", "Tensor", "The input tensor.")
-    .set_attr<FInferStructInfo>("FInferStructInfo", InferStructInfoCumsum);
+    .set_attr<FInferStructInfo>("FInferStructInfo", InferStructInfoCumsum)
+    .set_attr<Bool>("FPurity", Bool(true));
```

Comment:
will we run into issue if FPurity is not specified explicitly?

Reply:
We should specify it explicitly. I have an assert in the well-formedness checker that it's defined.